src/NUnitFramework/framework/Internal/Builders/PairwiseStrategy.cs

```diff
@@ -301,11 +302,11 @@ public bool IsTupleCovered( FeatureTuple tuple )
         /// </remarks>
         internal class PairwiseTestCaseGenerator
         {
-            private FleaRand _prng;
+            private FleaRand? _prng;
 
-            private int[] _dimensions;
+            private int[]? _dimensions;
 
-            private List<FeatureTuple>[][] _uncoveredTuples;
+            private List<FeatureTuple>[][]? _uncoveredTuples;
 
```

Comment:
Ok - let me check I'm understanding this correctly. You've used the `!` operator in this class, because these vars are created as null, but they're actually only used/accessible when you call `GetTestCases`, which initialises them all? (Which arguably also makes this a bit of a code smell, and suggests these values should be passed around as parameters, instead?)

Reply:
Yes, exactly! The `!` makes it more obvious where the previously implicit assumptions are.

Reply:
That makes them a nice target for refactoring later. I learned the hard way to add `!` first to show the world as it has been before doing that though, just to make it possible for reviewers to see what triggered the refactoring.

Reply:
I definitely agree with that approach. 🙂